prow/gerrit/adapter/trigger.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package adapter
+
+import (
+	"fmt"
+	"k8s.io/test-infra/prow/pjutil"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/gerrit/client"
+)
+
+const layout = "2006-01-02 15:04:05"
+
+func filterPresubmits(filter pjutil.Filter, change client.ChangeInfo, presubmits []config.Presubmit) ([]config.Presubmit, []config.Presubmit, error) {
```

Comment:
Shouldn't this be the one in PJUtil?

Reply:
Hmm, so this one still takes a gerrit change, similarly its counterpart depends on the githubclient.

Reply:
Can we abstract that?

Reply:
I would prefer to keep that out of scope for this PR. There is inevitably going to be some platform specific logic somewhere where we use the common filters in a certain way (e.g the way we get the changed files etc.).

Reply:
No, that is entirely within the scope of the PR. There is absolutely no reason to duplicate this logic. The reason Prow was so inconsistent in the past was that we did not allow for there to be one filter approach. The approach of:

 - filter, generating a matching value, a forcing function and a default
 - determine through a file changer whether the job Should Run
 - append to skip or not

 This is all common. Use your file listing adapter before you call this function, not inside of it, and re-use the common functionality.

Reply:
done